scipy/sparse/sparsetools.py

```diff
@@ -32,22 +32,22 @@ def _deprecated():
 # of rows.
 
 def csr_matvec(N, M, intptr, indices, data, x, y):
-    _sparsetools.csr_matvec(N, M, indptr, indices, data, 1, x, 0, y)
+    _sparsetools.csr_matvec(N, M, indptr, indices, data, 1, x, 1, y)
```

Comment:
Can you please explain why this change is correct?  Before these changes the contents of the `y` vector didn't matter.  Making these be `1` means that the values stored in the y vector before this call are now relevant.  This defeats the purpose of wrapping these to try and provide the old interface.